network/auth-header/src/getAuthHeadersFromConfig.ts

```diff
@@ -57,12 +57,16 @@ function splitKey (key: string) {
   return [key.slice(0, index), key.slice(index + 1)]
 }
 
-function loadToken (helperPath: string, settingName: string) {
+export function loadToken (helperPath: string, settingName: string) {
   if (!path.isAbsolute(helperPath) || !fs.existsSync(helperPath)) {
     throw new PnpmError('BAD_TOKEN_HELPER_PATH', `${settingName} must be an absolute path, without arguments`)
   }
 
-  const spawnResult = spawnSync(helperPath, { shell: true })
+  const isNode = helperPath.endsWith('.js')
+  const isWin = process.platform === 'win32'
+
+  // If it's windows and expected node, run with node
+  const spawnResult = spawnSync(isWin && isNode ? 'node' : helperPath, isWin && isNode ? [helperPath] : [], { shell: true })
```

Comment:
@zkochan Should I remove this logic and make another PR for it? It's just to be able to run JS helper tokens in Windows.

Reply:
yes